cmd/ledger/main.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"net/http"
+	_ "net/http/pprof"
+	"time"
+
+	"github.com/pingcap/tipocket/pkg/test-infra/tidb"
+	"github.com/pingcap/tipocket/pkg/verify"
+	"github.com/pingcap/tipocket/tests/ledger"
+
+	"github.com/pingcap/tipocket/cmd/util"
+	"github.com/pingcap/tipocket/pkg/cluster"
+	"github.com/pingcap/tipocket/pkg/control"
+	"github.com/pingcap/tipocket/pkg/test-infra/fixture"
+)
+
+var (
+	pprofAddr    = flag.String("pprof", "0.0.0.0:8080", "Pprof address")
+	namespace    = flag.String("namespace", "tidb-cluster", "test namespace")
+	nemesises    = flag.String("nemesis", "", "nemesis, separated by name, like random_kill,all_kill")
+	hub          = flag.String("hub", "", "hub address, default to docker hub")
+	imageVersion = flag.String("image-version", "latest", "image version")
+	storageClass = flag.String("storage-class", "local-storage", "storage class name")
+	// case config
+	accounts    = flag.Int("accounts", 1000000, "the number of accounts")
+	interval    = flag.Duration("interval", 2*time.Second, "check interval")
+	concurrency = flag.Int("concurrency", 200, "concurrency of worker")
+)
+
+func initE2eContext() {
+	fixture.E2eContext.LocalVolumeStorageClass = *storageClass
+	fixture.E2eContext.HubAddress = *hub
+	fixture.E2eContext.DockerRepository = "pingcap"
+	fixture.E2eContext.ImageVersion = *imageVersion
+}
+
+func main() {
+	flag.Parse()
+	initE2eContext()
+	go func() {
+		http.ListenAndServe(*pprofAddr, nil)
+	}()
+
+	cfg := control.Config{
+		Mode:        control.ModeSelfScheduled,
+		ClientCount: 1,
+		DB:          "noop",
+	}
+
+	provisioner, err := cluster.NewK8sProvisioner()
+	if err != nil {
+		log.Fatal(err)
+	}
+	suit := util.Suit{
+		Config:      &cfg,
+		Provisioner: provisioner,
+		ClientCreator: ledger.CaseCreator{Cfg: &ledger.Config{
+			NumAccounts: *accounts,
+			Concurrency: *concurrency,
+			Interval:    *interval,
```

Comment:
Do we need to set `TxnMode` here? May read from command flags

Reply:
I forget it.